### PR TITLE
feat: use etag and cache object response

### PR DESCRIPTION
**Pull request by preschian**
## Context

- Previously, there was no cache in the ipfs endpoint from image-workers. This PR is to add a cache to speed up the response time. Results can be seen here: https://github.com/kodadot/nft-gallery/pull/10320#issue-2304040793
- Reintroduce the cache again. Fix https://github.com/kodadot/workers/issues/220
- Add etag middleware as a fallback
- Add more ipfs gateways
- Cache response redirect (change status to 301; Chrome didn't cache 302 response)

## Ref

- [x] After - #288
- [x] https://github.com/kodadot/nft-gallery/discussions/10294

## PR Type

- [x] Feature

### CODE REVIEW

**Comment by preschian** [comment]
deploying to beta to see the results

**Comment by vikiival** [comment]
Test is failing

Screenshot 2024-05-21 at 11 21 11

**Comment by preschian** [comment]
description updated

**Comment by vikiival** [comment]
@preschian Is the test fixed?

File: services/image/src/routes/metadata.ts [hunk 1]
```diff
@@ -43,15 +43,19 @@ app.get('/*', async (c) => {
   }
 
   try {
+    // 0 redirect to internal url
+    if (url.includes('fxart-beta.kodadot.workers.dev')) {
+      return c.redirect(url, 301)
+    }
```
Comment (vikiival) [inline services/image/src/routes/metadata.ts:49]: and fxart prod?

**Review by vikiival — APPROVED** [review]

File: services/image/src/routes/metadata.ts (hunk 1 above)
Comment (preschian) [reply services/image/src/routes/metadata.ts]: updated

**Pull request merged**